mapdrawgdal.c

```diff
@@ -1983,6 +2040,16 @@ msDrawRasterLayerGDAL_16BitClassification(
 }
 
 /************************************************************************/
+/*                          IsNoData()                                  */
+/************************************************************************/
+static int
+IsNoData( double dfValue, double dfNoDataValue, int bIsNoDataNan )
+{
+  return (bIsNoDataNan && isnan(dfValue)) ||
+         (!bIsNoDataNan && dfValue == dfNoDataValue);
+}
```

Comment:
Hmm.. I thought about this more and yes your check is safer. If NoData is not NaN, my check returns No when dfValue is NaN (if that ever happens) because NaN == non-NaN is false. Logically, returning No is correct because dfValue (NaN) is different from non-NaN no data value, but proceeding with NaN dfValue further is not a good idea if dfValue gets ever assigned NaN for whatever reasons when NoData is not NaN.